data-prepper-plugins/sns-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/SNSSinkService.java

```diff
@@ -0,0 +1,241 @@
+
+    public static final String NUMBER_OF_RECORDS_FLUSHED_TO_SNS_SUCCESS = "snsSinkObjectsEventsSucceeded";
+
+    public static final String NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED = "snsSinkObjectsEventsFailed";
+
+    private static final String BUCKET = "bucket";
+
+    private static final String KEY_PATH = "key_path_prefix";
+
+    public static final String FIFO = ".fifo";
+
+    private final SNSSinkConfig snsSinkConfig;
+
+    private final Lock reentrantLock;
+
+    private final BufferFactory bufferFactory;
+
+    private final Collection<EventHandle> bufferedEventHandles;
+
+    private final SnsClient snsClient;
+
+    private final DlqPushHandler dlqPushHandler;
+
+    private final PluginSetting pluginSetting;
+
+    private Buffer currentBuffer;
+
+    private final int maxEvents;
+
+    private final ByteCount maxBytes;
+
+    private final long maxCollectionDuration;
+
+    private final String topicName;
+
+    private final int maxRetries;
+
+    private final Counter numberOfRecordsSuccessCounter;
+
+    private final Counter numberOfRecordsFailedCounter;
+
+    /**
+     * @param snsSinkConfig sns sink related configuration.
+     * @param bufferFactory factory of buffer.
+     * @param snsClient
+     * @param pluginMetrics metrics.
+     * @param pluginFactory
+     * @param pluginSetting
+     */
+    public SNSSinkService(final SNSSinkConfig snsSinkConfig,
+                          final BufferFactory bufferFactory,
+                          final SnsClient snsClient,
+                          final PluginMetrics pluginMetrics,
+                          final PluginFactory pluginFactory,
+                          final PluginSetting pluginSetting) {
+        this.snsSinkConfig = snsSinkConfig;
+        this.bufferFactory = bufferFactory;
+        this.snsClient = snsClient;
+        this.reentrantLock = new ReentrantLock();
+        this.bufferedEventHandles = new LinkedList<>();
+        this.maxEvents = snsSinkConfig.getThresholdOptions().getEventCount();
+        this.maxBytes = snsSinkConfig.getThresholdOptions().getMaximumSize();
+        this.maxCollectionDuration = snsSinkConfig.getThresholdOptions().getEventCollectTimeOut().getSeconds();
+        this.topicName = snsSinkConfig.getTopicArn();
+        this.maxRetries = snsSinkConfig.getMaxUploadRetries();
+        this.pluginSetting = pluginSetting;
+
+        this.numberOfRecordsSuccessCounter = pluginMetrics.counter(NUMBER_OF_RECORDS_FLUSHED_TO_SNS_SUCCESS);
+        this.numberOfRecordsFailedCounter = pluginMetrics.counter(NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED);
+
+        final Map<String, Object> pluginSettings = snsSinkConfig.getDlq()!= null ? snsSinkConfig.getDlq().getPluginSettings() : new HashMap<>();
+        String stsRoleARN = Objects.nonNull(pluginSettings.get("sts_role_arn")) ?  String.valueOf(pluginSettings.get("sts_role_arn")) : snsSinkConfig.getAwsAuthenticationOptions().getAwsStsRoleArn();
```

Comment:
Why are we checking for this in two different places? We should just be checking in the awsOptions config.

Reply:
Yes @graytaylor0 we have moved this condition checking to sink configuration.